api gw 1.16 updates (#18081)
hashicorp/consul: website/content/docs/api-gateway/configuration/index.mdx

- [GatewayClass](/consul/docs/api-gateway/configuration/gatewayclass) defines a class of gateway resources that you can use as a template for creating gateways.
- [GatewayClassConfig](/consul/docs/api-gateway/configuration/gatewayclassconfig) describes additional Consul API Gateway-related configuration parameters for the GatewayClass resource.
- [Routes](/consul/docs/api-gateway/configuration/routes) specifies the path from the gateway to the backend service(s) client to the listener.

> - [Intentions](/consul/docs/api-gateway/configuration/intentions) specifies traffic communication rules between Consul API Gateway and services in the Consul service mesh.

Reviewer: ```suggestion
- [Intentions](/consul/docs/connect/config-entries/service-intentions) specifies traffic communication rules between services in the mesh. You can also use them specify rules for communication between services outside the mesh that send traffic through Consul API gateway to services in the service mesh.
```
I made another comment directly on the proposed SI conf page, but just be clear, we only want one source of truth for configuration reference. In this case, it should be the existing SI conf entry ref.

Author: That's fair - I was torn between putting the existing link there and creating a new one. 

I'd like to include the intentions example somewhere, maybe in the upgrade docs, since this is a new workflow with 1.16. What do you think @trujillo-adam ?

Reviewer: The convention we started to implement is to add examples for specific use cases to the configuration reference page. https://developer.hashicorp.com/consul/docs/connect/config-entries/service-intentions#examples. The goal is to keep them all in one place as, well, a reference, and link to them as necessary. 
I wouldn't add the example to the upgrade docs, but it is something different that a user has consider when upgrading to 1.16. To that end we could mention that you need to configure intentions in those instructions and link to the example on the reference page.

You can create a basic Gateway object using the default [`gatewayClassName`](/consul/docs/api-gateway/configuration/gateway#gatewayclassname) (`consul`). If you want to create custom Gateways suitable for your environment, complete the following steps:

1. Define a [GatewayClassConfig](/consul/docs/api-gateway/configuration/gatewayclassconfig) that contains your custom configurations.
2. Define a [GatewayClass](/consul/docs/api-gateway/configuration/gatewayclass) and configure the [`parametersRef.name`](/consul/docs/api-gateway/configuration/gatewayclass#parametersref-name) to reference the name of your [GatewayClassConfig](/consul/docs/api-gateway/configuration/gatewayclassconfig).

> 3. Define a [Gateway](/consul/docs/api-gateway/configuration/gateway) and configure the [`gatewayClassName`](/consul/docs/api-gateway/configuration/gateway#gatewayclassname) to reference the name of your [GatewayClass](/consul/docs/api-gateway/configuration/gatewayclass).

Reviewer: @boruszak - "Define a GatewayClass and then configure the parametersRef.name to reference..." is correct, but it sounds strange in my ear. Probably the most technically accurate would be something like:

1. Create a GatewayClass configuration.
2. In the GatewayClass configuration, configure the `parametersRef.name` field to reference the name of your GatewayClassConfig file.

But I feel like we're splitting hairs. I'm fine with your suggestion.

> 3. Define a [Gateway](/consul/docs/api-gateway/configuration/gateway) and configure the [`gatewayClassName`](/consul/docs/api-gateway/configuration/gateway#gatewayclassname) to reference the name of your [GatewayClass](/consul/docs/api-gateway/configuration/gatewayclass).

Author: Oops - my IDE keeps auto-updating those numbers from repeated 1's to 1,2,3.. 

I haven't created a custom Gateway before, but from what I understand, defining all of these details and applying them together at once is necessary.

- [GatewayClass](/consul/docs/api-gateway/configuration/gatewayclass) defines a class of gateway resources that you can use as a template for creating gateways.
- [GatewayClassConfig](/consul/docs/api-gateway/configuration/gatewayclassconfig) describes additional Consul API Gateway-related configuration parameters for the GatewayClass resource.
- [Routes](/consul/docs/api-gateway/configuration/routes) specifies the path from the gateway to the backend service(s) client to the listener.

> - [Intentions](/consul/docs/connect/config-entries/service-intentions) specifies traffic communication rules between services in the mesh. You can also use them specify rules for communication between services outside the mesh that send traffic through Consul API gateway to services in the service mesh.

Reviewer: ```suggestion
- [Intentions](/consul/docs/connect/config-entries/service-intentions) specify traffic communication rules between services in the mesh. Intentions also enforce rules for service-to-service traffic routed through a Consul API gateway.
```
Sorry, I think this is better than my last suggestion
